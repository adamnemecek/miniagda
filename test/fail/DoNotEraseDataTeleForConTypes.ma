-- 2010-06-18

-- the following definition needs to be rejected!
data Wrap [A : Set] : Set
{ inn : (out : A) -> Wrap A
}
fields out

fun cast : [A : Set] -> [B : Set] -> A -> B
{ cast A B a = out {- B -} (inn {- A -} a)
}
