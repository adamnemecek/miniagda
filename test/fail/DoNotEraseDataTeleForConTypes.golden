MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "DoNotEraseDataTeleForConTypes.ma" ---
--- scope checking ---
--- type checking ---
type  Wrap : .[A : Set] -> Set
error during typechecking:
Wrap
/// constructor Wrap.inn
/// new Wrap : (.[A : Set] -> Set)
/// new A : Set
/// inferExpr' ^(out : A) -> Wrap A
/// inferExpr' A
/// inferExpr: variable A : Set may not occur
/// , because it is marked as erased
